src/renate/updaters/model_updater.py

```diff
@@ -40,28 +41,36 @@ def __init__(self, val_enabled: bool):
         super().__init__()
         self._report = Reporter()
         self._val_enabled = val_enabled
+        self._additional_metrics = AdditionalTrainingMetrics()
 
     @rank_zero_only
-    def _log(self, trainer: Trainer, training: bool) -> None:
+    def _log(self, trainer: Trainer, pl_module: LightningModule) -> None:
         """Report the current epoch's results to Syne Tune.
 
         If validation was run `_val_enabled` is True, the results are reported at the end of
         the validation epoch. Otherwise, they are reported at the end of the training epoch.
         """
 
+        training = pl_module.training
         if trainer.sanity_checking or (training and self._val_enabled):
             return
+        to_report = {k: v.item() for k, v in trainer.logged_metrics.items()}
+        to_report.update(self._additional_metrics(pl_module))
         self._report(
-            **{k: v.item() for k, v in trainer.logged_metrics.items()},
+            **to_report,
             step=trainer.current_epoch,
             epoch=trainer.current_epoch + 1,
         )
 
     def on_train_epoch_end(self, trainer: Trainer, pl_module: LightningModule) -> None:
-        self._log(trainer=trainer, training=pl_module.training)
+        self._additional_metrics.on_train_epoch_end()
+        self._log(trainer=trainer, pl_module=pl_module)
 
     def on_validation_epoch_end(self, trainer: Trainer, pl_module: LightningModule) -> None:
-        self._log(trainer=trainer, training=pl_module.training)
+        self._log(trainer=trainer, pl_module=pl_module)
+
+    def on_train_start(self, trainer: Trainer, pl_module: LightningModule) -> None:
```

Comment:
is this called also for sanity_checking?

Reply:
What is `this`?

Reply:
`on_train_start()`

Reply:
It runs on on_validation_start, hence the special handling checking for None in the Additional metrics class.